### PR TITLE
Fixes Wastage of screen real-estate on large displays

### PR DESCRIPTION
**Pull request by avgupt**
Fixes: https://issues.apache.org/jira/projects/CAMEL/issues/CAMEL-14769?filter=allopenissues

### CODE REVIEW

**Comment by zregvart** [comment]
Preview is [here](https://builds.apache.org/job/Camel.website/view/change-requests/job/PR-271/Preview/manual/latest/jmx.html). Judging form that the problem is not solved.

**Comment by avgupt** [comment]
Here's the preview

camel_website

**Comment by zregvart** [comment]
Preview is [here](https://deploy-preview-271--camel.netlify.com/)

**Comment by zregvart** [comment]
Also note the discussion previously had on #94.

File: antora-ui-camel/src/css/vars.css [hunk 1]
```diff
@@ -113,7 +113,7 @@
   /* static */
   --static-margin: 0 auto;
   /* frontpage */
-  --frontpage-max-width: calc(1140 / var(--rem-base) * 1rem);
+  --frontpage-max-width: 87.625vw;
```
Comment (zregvart) [inline antora-ui-camel/src/css/vars.css]: These are some very exact numbers how did you come by these?

File: antora-ui-camel/src/css/vars.css [hunk 2]
```diff
@@ -130,10 +130,10 @@
...
   --nav-panel-height--desktop: calc(var(--nav-height--desktop) - var(--drawer-height));
   --nav-width: calc(270 / var(--rem-base) * 1rem);
-  --doc-max-width: calc(720 / var(--rem-base) * 1rem);
-  --doc-max-width--desktop: calc(828 / var(--rem-base) * 1rem);
-  --static-max-width: calc(720 / var(--rem-base) * 1rem);
-  --static-max-width--desktop: calc(1140 / var(--rem-base) * 1rem);
+  --doc-max-width: 81vw;
+  --doc-max-width--desktop: 64vw;
+  --static-max-width: 81vw;
+  --static-max-width--desktop: 88vw;
```
Comment (zregvart) [inline antora-ui-camel/src/css/vars.css]: Previously we had this based on screen resolutions and the rem unit, not sure where these numbers come from, can you explain them?

**Comment by zregvart** [comment]
@avgupt are you working on this?

**Comment by avgupt** [comment]
@zregvart Yes I am, sorry for the delay.

File: antora-ui-camel/src/css/vars.css [hunk 3]
```diff
@@ -120,6 +120,7 @@
   --footer-font-color: var(--color-gray-70);
   --footer-link-font-color: var(--color-jet-80);
   /* dimensions */
+  --vw-base--desktop: calc(100 / 1300);   /* used to compute vw value from desired viewport size i.e. 1300 */
```
Comment (Delawen) [inline antora-ui-camel/src/css/vars.css:123]: Can I ask why 1300 is the desired viewport size? Is it limited by something else?

**Comment by Delawen** [comment]
On my 1920x1024 screen looks the preview looks good.

File: antora-ui-camel/src/css/vars.css [hunk 4]
```diff
@@ -131,9 +132,9 @@
   --nav-panel-height--desktop: calc(var(--nav-height--desktop) - var(--drawer-height));
   --nav-width: calc(270 / var(--rem-base) * 1rem);
   --doc-max-width: calc(720 / var(--rem-base) * 1rem);
-  --doc-max-width--desktop: calc(828 / var(--rem-base) * 1rem);
+  --doc-max-width--desktop: calc(828 * var(--vw-base--desktop) * 1vw);
   --static-max-width: calc(720 / var(--rem-base) * 1rem);
-  --static-max-width--desktop: calc(1140 / var(--rem-base) * 1rem);
+  --static-max-width--desktop: calc(1140 * var(--vw-base--desktop) * 1vw);
```
Comment (zregvart) [inline antora-ui-camel/src/css/vars.css:137]: This change in formulas seems arbitrary, what we had matches this logic: we want to have the width in vw. So to do that we divide the number in px by the base rem size, giving us the number of rems in a number of pixels, when we multiply that by 1 rem to get resulting the measure in vw.

You seem to multiply a number by a percentage and then multiply that by a device width varying unit.  Can you explain the logic behind the changed formula?

Comment (avgupt) [reply antora-ui-camel/src/css/vars.css:137]: I have calculated the variable --vw-base--desktop using a viewport width of 1300 (which we can change when we decide a standard value). As explained [here](https://stackoverflow.com/questions/28295072/how-can-i-convert-px-to-vw-in-javascript), the variable `--vw-base--desktop` stores the value of 1px in vw. To get the required max-width in vw, I multiplied the number in px by the variable `--vw-base--desktop`. Then I multiplied the number by 1vw to get the resulting measure in vw.

File: antora-ui-camel/src/css/vars.css (hunk 3 above)
Comment (avgupt) [reply antora-ui-camel/src/css/vars.css:123]: No, the desired viewport size is not limited by anything. Just a personal coice.

**Comment by zregvart** [comment]
What's missing here is the _why_. _Why_ is very important, we have a _why_ for the current width as discussed on #94. Outcome of that discussion is to have the `Wide / Narrow` functionality. So why change this now?

**Comment by avgupt** [comment]
The Wide functionality results in complete coverage of the website area which I believe, is not how the website is designed. This fix aims at handling the problem of wastage of screen real-estate on real-estate while retaining the design of the website i.e. the left and right margins. To the best of my understanding, the objective of Wide / Narrow functionality was to handle tables that are extra-large while this fix aims at resizing the max-width with respect to a viewport width.

**Comment by zregvart** [comment]
@avgupt Okay, lets try this approach, can you rebase this (seems that there are conflicts now). Seems that with this the Wide/Narrow functionality is no longer needed, would it make sense to remove it?

**Comment by zregvart** [comment]
Let me try to rebase this, with the #383 and the in page navigation panel on the right we need this.

**Comment by zregvart** [comment]
This was superseded by #386.

**Pull request closed without merging**